### PR TITLE
Raise warning for base_url ../embeddings .../completions .../rankings

### PR DESCRIPTION
**Pull request by raspawar**
#### Feature: UX: Raise warning for base_url ../embeddings .../completions .../rankings
Expected url format: **http://host:port**

### Current Implementation:
- Accept valid urls(/v1, ../v1/embeddings .../v1/completions .../v1/rankings) with UserWarning
- Error handling in case url does not follow `http://host:port` or above formats

### Implemented Warnings:
- For NvidiaDocumentEmbedder and NvidiaTextEmbedder `/v1` route is allowed and following warning is given for `v1/embeddings ` endpoint

      UserWarning: Expected format is 'http://host:port/'. Rest is ingnored.
- For NvidiaGenerator `/v1` route is allowed and following warning is given for `v1/chat/completions ` endpoint

      UserWarning: Expected format is 'http://host:port/'. Rest is ingnored.
- All other routes are rejected with ValueError:

      Base URL path is not recognized. Expected format is 'http://host:port/'. Rest is ingnored.

cc: @mattf @dglogo @sumitkbh

### CODE REVIEW

File: integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py [hunk 1]
```diff
@@ -86,6 +91,27 @@ def __init__(
...
+                pass
+            elif normalized_path in ["v1/embeddings"]:
+                warn_msg = f"{expected_format} Rest is ingnored."
+                warnings.warn(warn_msg, stacklevel=2)
+            else:
+                err_msg = f"Base URL path is not recognized. {expected_format}"
+                raise ValueError(err_msg)
+
+        base_url = urlunparse((result.scheme, result.netloc, "v1", "", "", ""))
+        return base_url
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py:113]: Looks like this code is mostly the same for all components? Let's move it into a `haystack_integrations.util.nvidia` module.

**Review by shadeMe — CHANGES_REQUESTED** [review]

File: integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py (hunk 1 above)
Comment (raspawar) [reply integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py:113]: Yes will do that

Comment (raspawar) [reply integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py]: @shadeMe can you take a look now?

File: integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import warnings
+from urllib.parse import urlparse, urlunparse
+
+
+def url_validation(api_url: str, default_api_url: str, allowed_paths: list):
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py:5]: Let's add docstrings for this function and use type hints from `typing`. Also, missing return type.

**Review by shadeMe — CHANGES_REQUESTED** [review]

File: integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py (hunk 2 above)
Comment (raspawar) [reply integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py]: I have resolved this, can you check again?

File: integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+    :param default_api_url:
+        The default API URL for comparison.
+    :param allowed_paths:
+        A list of allowed base paths that are valid if present in the URL.
+
+    :return:
+        A normalized version of the API URL with '/v1' path appended, if needed.
+
+    :raises ValueError:
+        If the base URL path is not recognized or does not match expected format.
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py:21]:
```suggestion
    :returns:
        A normalized version of the API URL with '/v1' path appended, if needed.
    :raises ValueError:
        If the base URL path is not recognized or does not match expected format.
```

File: integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    expected_format = "Expected format is 'http://host:port'."
+
+    if api_url == default_api_url:
+        return api_url
+    if result.path:
+        normalized_path = result.path.strip("/")
+        if normalized_path == "v1":
+            pass
+        elif normalized_path in allowed_paths:
+            warn_msg = f"{expected_format} Rest is ingnored."
```
Comment (shadeMe) [inline integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py:34]:
```suggestion
            warn_msg = f"{expected_format} Rest is ignored."
```

**Review by shadeMe — APPROVED** [review]
Just some minor changes. I'll merge once those are committed.

**Pull request merged**